packages/@aws-cdk/aws-ec2/lib/volume.ts

```diff
@@ -210,4 +215,470 @@ export enum EbsDeviceVolumeType {
-}
\ No newline at end of file
+
+  /**
+   * General purpose SSD volume that balances price and performance for a wide variety of workloads.
+   */
+  GENERAL_PURPOSE_SSD = GP2,
+
+  /**
+   * Highest-performance SSD volume for mission-critical low-latency or high-throughput workloads.
+   */
+  PROVISIONED_IOPS_SSD = IO1,
+
+  /**
+   * Low-cost HDD volume designed for frequently accessed, throughput-intensive workloads.
+   */
+  THROUGHPUT_OPTIMIZED_HDD = ST1,
+
+  /**
+   * Lowest cost HDD volume designed for less frequently accessed workloads.
+   */
+  COLD_HDD = SC1,
+
+  /**
+   * Magnetic volumes are backed by magnetic drives and are suited for workloads where data is accessed infrequently, and scenarios where low-cost
+   * storage for small volume sizes is important.
+   */
+  MAGNETIC = STANDARD,
+}
+
+/**
+ * An EBS Volume in AWS EC2.
+ */
+export interface IVolume extends IResource {
+  /**
+   * The EBS Volume's ID
+   *
+   * @attribute
+   */
+  readonly volumeId: string;
+
+  /**
+   * The availability zone that the EBS Volume is contained within (ex: us-west-2a)
+   */
+  readonly availabilityZone: string;
+
+  /**
+   * The customer-managed encryption key that is used to encrypt the Volume.
+   *
+   * @attribute
+   */
+  readonly encryptionKey?: IKey;
+
+  /**
+   * Grants permission to attach this Volume to an instance.
+   * CAUTION: Granting an instance permission to attach to itself using this method will lead to
+   * an unresolvable circular reference between the instance role and the instance.
+   * Use {@link IVolume.grantAttachVolumeToSelf} to grant an instance permission to attach this
+   * volume to itself.
+   *
+   * @param grantee  the principal being granted permission.
+   * @param instances the instances to which permission is being granted to attach this
+   *                 volume to. If not specified, then permission is granted to attach
+   *                 to all instances in this account.
+   */
+  grantAttachVolume(grantee: IGrantable, instances?: IInstance[]): Grant;
+
+  /**
+   * Grants permission to an instance to attach this Volume to itself.
+   *
+   * Note: This is implemented by adding a Tag with key `VolumeGrantAttach-<suffix>` to the
+   * instance and this Volume, and then conditioning the Grant on the presence of that Tag on
+   * both the instance and Volume. If you need to grant AttachVolume of this same Volume to
+   * multiple instances then provide a unique `tagKeySuffix` for each attachment; failure to do
+   * so will result in an inability to attach this volume to some instances.
+   *
+   * @param instance     The instance that will be allowed to attach this volume to itself.
+   * [disable-awslint:ref-via-interface]
+   * @param tagKeySuffix A suffix to use on the generated Tag key in place of the generated hash value.
+   *                     Defaults to a hash calculated from this volume.
+   */
+  grantAttachVolumeToSelf(instance: Instance, tagKeySuffix?: string): Grant;
```

Comment:
I'm thinking that I should change this, and similarly the corresponding `grantDetachVolumeFromSelf()`, to `grantAttachVolumeByTag(grantee: IGrantable, constructs: Construct[], tagKeySuffix?: string): Grant`

Reason being that there are other instance-sources that this will not cover as it's implemented -- ASGs, EC2-Fleets, and SpotFleetRequests to name a few.

Implementation would be, essentially, the same except that the tag value would be applied to all given constructs.